gas-tank/balance-services.md

```diff
@@ -0,0 +1,540 @@
+## Balance & Transactions
+
+### Get all balance entries for user
+
+This endpoint returns a list of all balance entries per address by using JWT
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances`, {
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+}).then((response) => {
+  console.log(response);
+  // Handle & do something with the response
+});
+```
+
+```json [Response]
+{
+  "additionalProp1": [
+    {
+      "id": "string",
+      "address": "string",
+      "tokenAddress": "string",
+      "balance": 0,
+      "chain": "string"
+    }
+  ],
+  "additionalProp2": [
+    {
+      "id": "string",
+      "address": "string",
+      "tokenAddress": "string",
+      "balance": 0,
+      "chain": "string"
+    }
+  ],
+  ...
+}
+```
+
+:::
+
+### Get balances by address
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/${address}`, { // Address to retrieve balances // [!code highlight]
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json"
+    "Authorization": `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+})
+  .then((response) => {
+    console.log(response);
+    // Handle & do something with the response
+  })
+```
+
+```json [Response]
+[
+  {
+    "id": "string",
+    "address": "string",
+    "tokenAddress": "string",
+    "balance": 0,
+    "chain": "string"
+  },
+  ...
+]
+```
+
+:::
+
+### Get total balances by JWT
+
+This endpoint returns a list of total balances for the specified JWT.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/totals`, {
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+}).then((response) => {
+  console.log(response);
+  // Handle & do something with the response
+});
+```
+
+```json [Response]
+[
+  {
+    "address": "string",
+    "symbol": "string",
+    "value": "string",
+    "decimals": 0
+  },
+  ...
+]
+```
+
+:::
+
+### Increase balance
+
+This endpoint allows users to increase their balance on the Gas Tank platform.
```

Comment:
Can you say in other words it's for depositing tokens; can we add a table of the confirmation count for it ?

cc. @VachaganGrigoryan